File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.py
class Solution:
    def maxProduct(self, nums: List[int]) -> int:
        n = len(nums)
        largest = nums[0]
        second_largest = None
        
        for i in range(1, n):
            if nums[i] > largest:
                second_largest = largest
                largest = nums[i]
            elif second_largest is None or nums[i] > second_largest:
                second_largest = nums[i]
        
        return (largest - 1) * (second_largest - 1)

File: 1464-maximum-product-of-two-elements-in-an-array/1464-maximum-product-of-two-elements-in-an-array.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxProduct = function(nums) {
    let n = nums.length;
    let largest = nums[0];
    let secondLargest;
    
    for (let i = 1; i < n; i++){
        if (nums[i] > largest){
            secondLargest = largest;
            largest = nums[i];
        }
        else if (!secondLargest || nums[i] > secondLargest){
            secondLargest = nums[i];
        }
    }
    
    return (largest - 1) * (secondLargest - 1);
};